### PR TITLE
[REPLICATED-1805] custom preflight checks

### PR DESCRIPTION
**Pull request by emosbaugh**

### CODE REVIEW

File: validate.go [hunk 1]
```diff
@@ -827,11 +835,6 @@ func ContainerNameExists(v *validator.Validate, topStruct reflect.Value, current
 		return true
 	}
 
-	if hasReplTemplate(field) {
```
Comment (divolgin) [inline validate.go:830]: This seems unrelated.  Why this change?

**Review by divolgin — APPROVED** [review]

File: validate.go (hunk 1 above)
Comment (emosbaugh) [reply validate.go:830]: oops. i will remove
